### DISCUSSION TITLE
Cloud based services for running RemoteApps

### DISCUSSION THREAD

**Discussion by ohault** [kimmknight/remoteapptool#146] [Q&A]
Does exist Cloud based services for running RemoteApp applications?

A very basic use case would be, a simple standalone win32 app with GUI without any user file operations to perform, like calc.exe

**Comment by kimmknight** [comment]
I've heard from a number of people using cloud-hosted machines to host RemoteApps (using RemoteApp Tool). Both Azure and AWS VMs.

Microsoft offer their [Azure Virtual Desktop](https://learn.microsoft.com/en-us/azure/virtual-desktop/overview) service - I'm don't have any experience with this, so I'm unsure if you can host an individual machine, or whether you need other infrastructure.

10 years ago, I hosted RemoteApps on a single Azure VM - it worked quite well. I have also since hosted an RD server on [ChicagoVPS](https://www.chicagovps.net/services/cloud-vps) as their prices were pretty good. This was for a temporary project, and I recklessly just exposed the RDP port to the internet (although I changed it from 3389 to something obscure). It was hacked within 3 weeks. In my case this didn't really matter, but serves as a good lesson.

So, if you are going to cloud-host an RD server, I would suggest the following:

- Don't expose the RDP port to the internet, setup a Remote Desktop Gateway (either with Windows Server, or using [rdpgw](https://github.com/bolkedebruin/rdpgw))
- Make sure the cloud server is in your region. If you don't, the lag can be quite problematic for RD sessions.

**Comment by ohault** [comment]
Do you know server-less technologies like Azure Functions or Cloudflare Workers ?

The idea is that this same concept could be apply to the server part of RemoteApp in place a full desktop on a dedicated server.

The approach could become entirely dynamic and initiated from the desktop

Right click on an exe/app and select « Run on … » , the user will select an environnement from a list (like Windows Sandbox, Docker Desktop, or a cloud app runner provider)

A cloud app runner provider would offer a fully dynamic and server-less env for running gui apps streamed back to the user device.

**Comment by kimmknight** [comment]
That's a cool idea - well beyond my expertise to develop!
